rust/value.rs

```diff
@@ -0,0 +1,130 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::fmt;
+
+use crate::{
+    common::{error::TypeQLError, Span, Spanned},
+    pretty::Pretty,
+    Result,
+};
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Category {
+    Boolean,
+    Integer,
+    Decimal,
+    Date,
+    DateTime,
+    DateTimeTZ,
+    Duration,
+    String,
+}
+
+impl Category {
+    fn name(&self) -> &'static str {
+        match self {
+            Self::Boolean => "Boolean",
+            Self::Integer => "Integer",
```

Comment:
`Long`?
Is Double going to be present?

Reply:
No, because we can't distinguish a Double from a Decimal at this stage. Integer literal may be a Long, a Double, or a Decimal.
I'd rather rename Decimal to smth else (Fraction? kinda implies a ratio though) than Integer to Long, so there's less presumption of the actual value type.